components/tidb_query_vec_expr/src/impl_string.rs

```diff
@@ -425,6 +425,27 @@ pub fn hex_str_arg(arg: Option<BytesRef>) -> Result<Option<Bytes>> {
     Ok(arg.map(|b| hex::encode_upper(b).into_bytes()))
 }
 
+#[inline]
+fn find_str(text: &str, pattern: &str) -> Option<usize> {
+    twoway::find_str(text, pattern).map(|i| text[..i].chars().count())
+}
+
+#[rpn_fn(nullable)]
+#[inline]
+pub fn locate_2_args_utf8(substr: Option<BytesRef>, s: Option<BytesRef>) -> Result<Option<i64>> {
+    let (substr, s) = match (substr, s) {
+        (Some(v1), Some(v2)) => match (str::from_utf8(v1), str::from_utf8(v2)) {
+            (Ok(s1), Ok(s2)) => (s1, s2),
+            _ => return Ok(None),
```

Comment:
```suggestion
            (Err(err), Ok(_)) => return Err(box_err!("invalid substr: {:?}", err)),
            (Ok(_), Err(err)) => return Err(box_err!("invalid s: {:?}", err)),
            (Err(err1), Err(err2)) => return Err(box_err!("invalid s and substr: {:?} {:?}", err1, err2)),
```

Should I do this for proper error handling?

Reply:
Returning `None` is OK